src/systems/battery_plugin/LinearBatteryPlugin.cc

```diff
@@ -439,10 +480,26 @@ void LinearBatteryPlugin::PreUpdate(
 {
   IGN_PROFILE("LinearBatteryPlugin::PreUpdate");
 
-  // \todo(anyone) Add in the ability to stop the battery from draining
-  // after it has been started by a topic. See this comment:
-  // https://github.com/ignitionrobotics/ign-gazebo/pull/1255#discussion_r770223092
-  this->dataPtr->startDraining = this->dataPtr->startDrainingFromTopics;
+  // Recaculate the toal power load among consumers
+  double total_power_load = this->dataPtr->initialPowerLoad;
+  _ecm.Each<components::BatteryPowerLoad>(
+    [&](const Entity & /*_entity*/,
+        const components::BatteryPowerLoad *_batteryPowerLoadInfo)->bool
+    {
+      if (_batteryPowerLoadInfo->Data().batteryId ==
+          this->dataPtr->batteryEntity)
+      {
+        total_power_load = total_power_load +
+            _batteryPowerLoadInfo->Data().batteryPowerLoad;
+      }
+      return true;
+    });
+
+  bool success = this->dataPtr->battery->SetPowerLoad(
+      this->dataPtr->consumerId, total_power_load);
+  if (!success)
+      ignerr << "Failed to set consumer power load." << std::endl;
+
   // Start draining the battery if the robot has started moving
```

Comment:
Does this block make sense with the new architecture? I mean, it seems that we're checking if the robot has some velocity in order to start draining the battery. Should that now be taken care by each battery consumer?

Reply:
I think it's OK to leave it as it is for now for not changing behavior.